src/file_pipe_log/format.rs

```diff
@@ -142,6 +184,15 @@ impl LogFileHeader {
     }
 }
 
+/// Basic trait for the successive LogXXX.
+pub trait Header {
```

Comment:
In my perspective, the `LogFileWriter` and `LogFileReader` are accessors to LogFile for writing and reading. And each of them holds a `header` to represent the metadata for the related log file header. So, the `header` is proposed to provide the basic api for accessing and updating the `header` of `LogFileXXX`.

Reply:
I think you can directly access the header field instead of using a method? Abstraction is only needed if there are some other unknown types that might implement this trait.

Reply:
Yep, i just wanna import the `Header` as a common trait for `LogFileReader` and `LogFileWriter`. But it seems that it's not an appropriate design.
Your suggestion is a better choice for this.

Reply:
Resolved